app/controllers/spree/adyen_notifications_controller.rb

```diff
@@ -5,16 +5,15 @@ class AdyenNotificationsController < AdyenController
     before_action :authenticate
 
     def notify
-      if notification_exists?(params)
-        accept
-      else
-        notification = AdyenNotification.build(params)
-        notification.save!
+      notification = AdyenNotification.build(params)
+      notification.save!
 
-        # prevent alteration to associated payment while we're handling the action
-        Spree::Adyen::NotificationProcessor.new(notification).process!
-        accept
-      end
+      # prevent alteration to associated payment while we're handling the action
+      Spree::Adyen::NotificationProcessor.new(notification).process!
+      accept
+    rescue ActiveRecord::RecordNotUnique
```

Comment:
Can we move this to be a begin-rescue that only covers the call to `save!`?  That's a bit safer and makes the handling a bit more obvious I think?